lib/simperium/bucket-store.js

```diff
@@ -1,113 +1,84 @@
-var Promise = require('promise');
+class StoreProvider {
+	constructor( config ) {
+		this.setup = new Promise( ( resolve, reject ) => config( resolve, reject ) );
```

Comment:
sure, though I didn't want to change any functionality in this PR. I think that would require making changes at the calling sites - a good change for a future PR